mozilla_aws_cli/sts_conn.py

```diff
@@ -43,20 +48,31 @@ def get_credentials(bearer_token, id_token_dict, role_arn):
 
             # Call the STS API
             resp = requests.get(url=sts_url, params=parameters)
+            try:
+                root = ElementTree.fromstring(resp.content)
+            except ElementTree.ParseError as e:
+                raise MalformedResponseWarning('Unable to parse XML response to AssumeRoleWithWebIdentity call')
             if resp.status_code != requests.codes.ok:
-                if 'The requested DurationSeconds exceeds the MaxSessionDuration set for this role' in resp.text:
+                error = dict(
+                    [(x.tag.split('}', 1)[-1], x.text) for x in root.find(
```

Comment:
`ElementTree` queries are pretty hard to understand, I would probably add a comment here to explain what is happening.

Reply:
Ya, I've got an issue ( #23 ) to de-objuscate this. I'd intended to here but couldn't figure out a clearer way to say it. I'll keep trying.